tests/foreman/cli/test_hostgroup.py

```diff
@@ -151,6 +157,173 @@ def test_positive_create_with_puppet_proxy(self):
+        arch = 'x86_64'
+        hostgroup = make_hostgroup({'architecture': arch})
+        self.assertEqual(
+            arch,
+            hostgroup['architecture'],
+        )
+
+    @run_only_on('sat')
+    @tier1
+    def test_positive_create_with_domain(self):
+        """Check if hostgroup with domain can be created
+
+        @id: c468fcac-9e42-4ee6-a431-abe29b6848ce
+
+        @Assert: Hostgroup should be created and has domain assigned
+
+        """
+        domain = make_domain()
+        hostgroup = make_hostgroup({'domain-id': domain['id']})
+        self.assertEqual(
+            domain['name'],
+            hostgroup['domain'],
+        )
+
+    @skip_if_bug_open('bugzilla', 1309107)
+    @run_only_on('sat')
+    @tier1
+    def test_positive_create_with_lifecycle_environment(self):
+        """Check if hostgroup with lifecyle environment can be created
+
+        @id: c468fcac-9e42-4ee6-a431-abe29b6848ce
+
+        @Assert: Hostgroup should be created and has lifecycle env assigned
+
+        @BZ: 1309107
+        """
+        org = make_org()
+        lc_env = make_lifecycle_environment({'organization-id': org['id']})
+        hostgroup = make_hostgroup({
+            'lifecycle-environment': lc_env['name'],
+            'organization-ids': org['id'],
+            'lifecycle-environment-organization-id': org['id'],
+        })
+        self.assertEqual(
+            lc_env['name'],
+            hostgroup['lifecycle-environment'],
+        )
+
+    @skip_if_bug_open('bugzilla', 1309107)
+    @run_only_on('sat')
+    @tier1
+    def test_positive_create_with_multiple_entities(self):
+        """Check if hostgroup with multiple options can be created
+
+        @id: a3ef4f0e-971d-4307-8d0a-35103dff6586
+
+        @Assert: Hostgroup should be created and has all defined entities
+        assigned
+
+        @BZ: 1309107
+        """
+        subnet = make_subnet()
+        domain = make_domain()
+        org = make_org()
+        lc_env = make_lifecycle_environment({'organization-id': org['id']})
+        loc = make_location()
+        env = make_environment()
+        os = make_os()
+        media = make_medium()
+        arch = make_architecture()
+        puppet_proxy = Proxy.list()[0]
+        ptable = 'Kickstart default'
+        hostgroup = make_hostgroup({
+            'domain-id': domain['id'],
+            'subnet-id': subnet['id'],
+            'organization-ids': org['id'],
+            'location-ids': loc['id'],
+            'environment-id': env['id'],
+            'puppet-proxy-id': puppet_proxy['id'],
+            'puppet-ca-proxy-id': puppet_proxy['id'],
+            'operatingsystem-id': os['id'],
+            'architecture-id': arch['id'],
+            'lifecycle-environment': lc_env['name'],
+            'lifecyle-environment-organization-id': org['id'],
+            'partition-table': ptable,
+            'medium-id': media['id'],
+        })
+        self.assertEqual(domain['name'], hostgroup['domain'])
```

Comment:
You created so many entities, but have only 4 assertions, I would like the idea to have same amounts of entities and assertions whether less of one or more of last